### ISSUE TITLE
EWSTimeZone.localzone() crash on Unix when env var TZ=UTC is set

### ISSUE THREAD

**Issue by salty-horse** [ecederstrand/exchangelib#1111]
**Describe the bug**
When the `TZ=UTC` environment variable is set on Unix environments, calls to `EWSTimeZone.localzone()` crash.
This is because `tzlocal.get_localzone()` returns a wrapped native python UTC object `_PytzShimTimezone(datetime.timezone.utc, 'UTC')` that doesn't have a `.key`, which exchangelib expects.

**To Reproduce**
Run this command with exchangelib installed:
```
$ TZ=UTC python -c 'from exchangelib import EWSTimeZone; EWSTimeZone.localzone()'

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/exchangelib/ewsdatetime.py", line 291, in localzone
    return cls.from_timezone(tz)
  File "/exchangelib/ewsdatetime.py", line 272, in from_timezone
    return {
  File "/exchangelib/ewsdatetime.py", line 278, in <lambda>
    'pytz_deprecation_shim': lambda z: cls.from_zoneinfo(z.unwrap_shim())
  File "/exchangelib/ewsdatetime.py", line 264, in from_zoneinfo
    return cls(tz.key)
AttributeError: 'datetime.timezone' object has no attribute 'key'
```

**Expected behavior**
No crash.

**Additional context**
For example, Python and exchangelib versions.
Using exchangelib 4.6.0, tzlocal 4.1.0

Perhaps it is a `tzlocal` bug? Does it promise to always return a `.key`-able timezone?

The correct fix would probably for exchangelib to compare with `datetime.timezone.utc` before trying `self.tzinfo.key == "UTC"`

**Comment by ecederstrand** [comment]
This was fixed in https://github.com/ecederstrand/exchangelib/commit/936c134c2724b090c63f18f1b3fbee6fae7a2d8b, available from version 4.6.1

**Comment by salty-horse** [comment]
Sorry, I totally missed I wasn't using the latest version! Thanks :)
